openreview/tools.py

```diff
@@ -36,19 +36,34 @@ def create_profile(client, email, first, last, middle = None, allow_duplicates =
 
     if not profile:
 
-        # validate only first and last names.
-        # this is so that we catch more potential collisions.
+        # validate the name with just first and last names,
+        # and also with first, middle, and last.
+        # this is so that we catch more potential collisions;
         # let the caller decide what to do with false positives.
-        validation_response = client.get_tildeusername(first, last, None)
 
-        # the username in validation_response will end with 1 if no other profiles with this first and last name exist.
-        if validation_response['username'].endswith('1'):
+        username_response_FL_only = client.get_tildeusername(
+            first,
+            last,
+            None
+        )
+
+        username_response_full = client.get_tildeusername(
+            first,
+            last,
+            middle
+        )
+
+        # the username in each response will end with 1
+        # if profiles don't exist for those names
+        username_FL_unclaimed = username_response_FL_only['username'].endswith('1')
+        username_full_unclaimed = username_response_full['username'].endswith('1')
+
+        if all([username_FL_unclaimed, username_full_unclaimed]):
             profile_exists = False
         else:
             profile_exists = True
 
-        id_response = client.get_tildeusername(first, last, middle)
-        tilde_id = id_response['username']
+        tilde_id = username_response_full['username']
```

Comment:
If a profile with name fname=A, lname=C (tilde = ~A_C_1) already exists, and the user tries to create a new profile with fname=A, mname=B, and lname=C, and say allow_duplicates=True, then the tilde id that the user gets is ~A_B_C_1 and not ~A_C_2. Is this ok ?

Reply:
I think that's the expected behavior. If a middle name is passed into create_profiles, it should try to create a profile with the middle name included.